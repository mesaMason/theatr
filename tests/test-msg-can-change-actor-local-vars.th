dolphin(int weight, int age):
    print("inside dolphin.")
    int sum = weight + age

    print("sum = ")
    print(sum)

    receive:
        swim():
            print("inside swim().")
            print("sum = ")
            print(sum)
        eat():
            print("inside eat().")
            print("sum = ")
            sum = sum + 5
            print(sum)
    drop:
        print("inside drop().")
        print("sum = ")
        sum = sum + 5
        print(sum)
    after:
        return;
        
func main() -> int:
    int weight = 10
    int age = 3
    actor d = new dolphin(weight, age)
    sadfdsaf() | d
    eat() | d
    swim() | d
    die() | d
    return 0

