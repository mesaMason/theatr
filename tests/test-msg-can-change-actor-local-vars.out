new dolphin created.
inside dolphin.
sum = 
13
inside drop().
sum = 
18
inside eat().
sum = 
23
inside swim().
sum = 
23
